sdk/core/core/src/az_log.c

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_log.h>
+
+#include <_az_cfg.h>
+
+static az_log_classification const * _az_log_classifications = NULL;
+static size_t _az_log_classifications_length = 0;
+static az_log * _az_log_listener = NULL;
+
+void az_log_set_classifications(
+    az_log_classification const * const classifications,
+    size_t const classifications_length) {
+  _az_log_classifications = classifications;
+  _az_log_classifications_length = classifications_length;
+}
+
+void az_log_reset_classifications() {
+  _az_log_classifications = NULL;
+  _az_log_classifications_length = 0;
+}
+
+void az_log_set_listener(az_log * const listener) { _az_log_listener = listener; }
+
+void az_log_write(az_log_classification const classification, char const * const message) {
```

Comment:
IMPORTANT: We'll probably need to add thread safety to ALL these functions.

Reply:
Yes, the char* should be a span

Reply:
Done char => span

Reply:
I put the TODO comments, and created https://github.com/Azure/azure-sdk-for-c/issues/316
I consider I've done your suggestion in scope of this PR.